engine/classes/Elgg/Database/MetadataTable.php

```diff
@@ -123,7 +126,10 @@ function create($entity_guid, $name, $value, $value_type = '', $owner_guid = 0,
 		}
 	
 		$access_id = (int) $access_id;
-	
+		
+		if (strlen($value) >= self::MAX_VALUE_LENGTH) {
+			elgg_log($name, 'WARNING');
```

Comment:
Better warning: `"Metadata '$name' is above the MySQL TEXT size limit and may be truncated."`